config/logstash.yml

```diff
@@ -165,6 +165,15 @@
 # api.auth.basic.username: "logstash-user"
 # api.auth.basic.password: "s3cUreP4$$w0rD"
 #
+#
+# Applies when `api.auth.type` `basic`.
+# Sets Logstash to WARN or ERROR when HTTP authentication password doesn't meet the password policy requirements.
+# The default is WARN. Setting to ERROR enforces stronger passwords (recommended).
+# The default password policy requires non-empty minimum 5 char string that includes a digit, upper case letter and lower case letter.
+# Set `api.auth.basic.password_policy.include.symbol` `REQUIRED` to accept only passwords that include at least one special character.
```

Comment:
I am _not_ in favor of turning this part of the policy on by default.

> ### Password length is more important than password complexity
> 
> NIST has moved away from password complexity and now recommends longer passwords. Enforcing complex passwords that contain upper- and lower-case letters, numbers, and special characters will ensure strong passwords are created in theory, but in practice, these requirements result in weak passwords being created – Password123! for instance, would meet complexity requirements but is not a strong password. Instead, encourage the use of passphrases and set the maximum password field length at 64 characters. Password length, character for character, is more important than password complexity.
>
> -- [NetSec News Summary of NIST Special Publication 800-63B](https://www.netsec.news/summary-of-the-nist-password-recommendations-for-2021/)

Reply:
Comment for 1,2 & 3 points.
1. Agree with it from Elastic stack point of view.
2. Provided detail explanations in the docs, plz take a look.
3.  TBH, I would be still supportive if we separate password policy and use it in whenever cases LS requires (eg., HTTP basic, SSL keystore, etc...). We can step-by-step extend this since core is well abstracted.

@yaauie I did quick research before implementing this feature and figured out every organization (AWASP, NIST, etc..) has their own point of view for stronger passwords. And for our case, IMHO, we are perfectly providing a control over policy according to organization's rules and providing basic requirement to lower the security risk.